cli/src/commands/mod.rs

```diff
@@ -2141,7 +2144,24 @@ fn cmd_commit(ui: &mut Ui, command: &CommandHelper, args: &CommitArgs) -> Result
     let matcher = workspace_command.matcher_from_values(&args.paths)?;
     let mut tx = workspace_command.start_transaction(&format!("commit {}", commit.id().hex()));
     let base_tree = merge_commit_trees(tx.repo(), &commit.parents())?;
-    let tree_id = tx.select_diff(ui, &base_tree, &commit.tree()?, matcher.as_ref(), "", false)?;
+    let instructions = format!(
+        "\
+You are making a partial commit from the changes in: {}
+
+The diff initially shows all changes. Adjust the right side until it shows the
+contents you want for the new commit. The remainder will stay in the working
+copy.
+",
+        tx.format_commit_summary(&commit)
```

Comment:
Since this is always going to describe the working-copy commit, I'm not sure how useful it's going to be to include the commit summary line. I don't really mind including it if you want it.

Reply:
I could see it being useful for someone who pre-describes their commits with a message like `wip: foo`, as a reminder for what changes to include. We can remove it later if we decide it's not useful.